modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc

```diff
@@ -303,17 +304,20 @@ CONCAT_WS(''," . $DB->quote($dir_path) . ",'trashbin/',SUBSTRING_INDEX(minc_loca
             'minYear'         => $study['startYear'] - $study['ageMax'],
             'maxYear'         => $study['endYear'] - $study['ageMin']
         );
-        $problemOptions = array(
-            '' => 'Any',
-            'CandID and PSCID do not match database'
-            => 'Candidate Mismatch',
-            'Could not identify scan type'
-            => 'Could not identify scan type',
-            'Protocol Violation'
-            => 'MRI Protocol Check violation',
-            'Visit label does not exist'
-            => 'Visit label does not exist'
+
+        $problemOptions = ['' => 'Any'];
+
+        // Build a list of existing problem types based on data in db
+        $problemTypes = $db->pselect(
```

Comment:
I think this is a case where denormalization is fine, because we want to be able to insert arbitrary errors without modifying the setup, and it's better to have inconsistent error messages than errors that get lost because of foreign key constraints.